plugins/engine-datafusion/jni/src/row_id_optimizer.rs

```diff
@@ -25,6 +26,134 @@ use datafusion::physical_plan::projection::ProjectionExec;
 #[derive(Debug)]
 pub struct FilterRowIdOptimizer;
 
+impl FilterRowIdOptimizer {
```

Comment:
we are using unwrap at lot of places in this function. let's move it to .expect instead?